File: templates/Uwp/_composition/MVVMLight/Feature.DeepLinking_TabbedPivotVB/Activation/SchemeActivationHandler_postaction.vb
'{[{
Imports Param_ItemNamespace.Views
'}]}
Namespace Activation    
    Friend Class SchemeActivationHandler        
        Inherits ActivationHandler(Of ProtocolActivatedEventArgs)
        '{[{

        Private ReadOnly Property NavigationService As NavigationServiceEx
            Get
                Return CommonServiceLocator.ServiceLocator.Current.GetInstance(Of NavigationServiceEx)()
            End Get
        End Property

        ' By default, this handler expects URIs of the format 'wtsapp:sample?paramName1=paramValue1&paramName2=paramValue2'
        Protected Overrides Async Function HandleInternalAsync(args As ProtocolActivatedEventArgs) As Task
            ' Create data from activation Uri in ProtocolActivatedEventArgs
            Dim data = New SchemeActivationData(args.Uri)
            If data.IsValid Then
                Dim frame = TryCast(Window.Current.Content, Frame)
                Dim pivotPage = TryCast(frame.Content, PivotPage)
                If pivotPage IsNot Nothing Then
                    Await pivotPage.InitializeFromSchemeActivationAsync(data)
                Else
                    NavigationService.Navigate(GetType(ViewModels.PivotViewModel).FullName, data)
                End If
            ElseIf args.PreviousExecutionState <> ApplicationExecutionState.Running Then
                NavigationService.Navigate(GetType(ViewModels.PivotViewModel).FullName)
            End If
            Await Task.CompletedTask
        End Function
        '}]}
    End Class
End Namespace

File: templates/Uwp/_comp/CaliburnMicro/Feat.DeepLinking_Blank/Activation/SchemeActivationHandler_postaction.cs
//{[{
using Caliburn.Micro;
using Param_RootNamespace.ViewModels;
//}]}
namespace Param_RootNamespace.Activation
{
    internal class SchemeActivationHandler : ActivationHandler<ProtocolActivatedEventArgs>
    {
//{[{
        private INavigationService _navigationService;

        public SchemeActivationHandler(INavigationService navigationService)
        {
            _navigationService = navigationService;
        }

        // By default, this handler expects URIs of the format 'wtsapp:sample?paramName1=paramValue1&paramName2=paramValue2'
        protected override async Task HandleInternalAsync(ProtocolActivatedEventArgs args)
        {
            // Create data from activation Uri in ProtocolActivatedEventArgs
            var data = new SchemeActivationData(args.Uri);
            if (data.IsValid)
            {
                _navigationService.Navigate(data.PageType, data.Parameters);
            }

            await Task.CompletedTask;
        }

//}]}
    }
}
